### PR TITLE
add HelloWorld-simple project

### PR DESCRIPTION
**Pull request by ChugR**
Here is the first of two Hello World examples. HelloWorld-simple compares with the examples delivered with [Apache Qpid clients](https://qpid.apache.org/components/messaging-api/index.html#examples). New users can get started by taking  only a shallow dive into AMQP.

This example enjoys a fairly complete line-by-line dissection in my [blog post](https://chugrolke.wordpress.com/2015/08/03/hello-world/).

-Chuck

### CODE REVIEW

File: Examples/HelloWorld/HelloWorld-simple/HelloWorld-simple.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+﻿//  ------------------------------------------------------------------------------------
+//  Copyright (c) Chuck Rolke
```
Comment (xinchen10) [inline Examples/HelloWorld/HelloWorld-simple/HelloWorld-simple.cs]: //  Copyright (c) Microsoft Corporation

Microsoft OSS projects require the above Copyright line. Without that we cannot accept the contribution.

File: Examples/HelloWorld/HelloWorld-simple/HelloWorld-simple.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//   messages through the endpoint address. 
+//
+// Tested with:
+//    * Amqpnetlite TestAmqpBroker
+//    * Apache ActiveMQ 5.12
+//    * Apache Qpidd 0.34
+//
+// Description:
+//   Please see https://chugrolke.wordpress.com/2015/08/03/hello-world/
+//   for a discussion of how this code works.
```
Comment (xinchen10) [inline Examples/HelloWorld/HelloWorld-simple/HelloWorld-simple.cs]: Can you include a README.md file to have the information right here with the code so we don't need to link to external web sites?

**Comment by ChugR** [comment]
Thanks for the review. Fixing for later submission.

**Comment by ChugR** [comment]
It doesn't make sense for me to submit code with Copyright Microsoft in it. Looking at other Azure projects one can see plenty of non-Microsoft copyrights:

- azure-powershell: C 2006, 2008 Junio C Hamano
- sdk-for-go: C 2015 The Go Authors (dozens of files)
- linux-automation: C John D. McCalpin

even amqpnetlite has some Copyright Red Hat already!

I've signed a CLA. I've included Apache Licence, Version 2.0 boiler plate in all sources. We should be good to go.

I am not a lawyer but the moment I make a submission the copyright is mine whether I put a copyright statement in the source or not. Would the submission be acceptable if I changed all the sources to include NO copyright statements?

-Chuck

**Comment by jeffwilcox** [comment]
Chuck,
Thanks for signing the CLA and for proposing your contribution. The contribution you're proposing is yours, but as it is direct contribution to this project (you're submitting a pull request and have signed a CLA), we would like to try and keep the files consistent. This is pretty standard practice for open source projects when receiving a contribution directly from a collaborator or else over time projects would get very convoluted on a per-file basis as years of names were added.

In the examples you've provided, those pieces of code inside the open source project came from other projects, and so we have to include their licenses, but it requires us to work with our legal teams to make sure that everything is clean and ready to go with those contributions. It's a lot more work.

If we had instead gone out and found this open source code in a compatibly-licensed project elsewhere and worked with our team to get the OK to include that other project's code, then we would of course keep the copyright headers and any required notices in place, since that's important to OSS.

Here's an example of what the [Apache Foundation requires](http://www.apache.org/legal/src-headers.html#headers): they ask that the contributor not include their own copyright statement, and if they so wish, they may use a different file like CONTRIBUTORS or NOTICE to include the credit if the person would like to be included in the source code. This then helps their tooling to understand that the contribution was direct to the project and not through the inclusion of other works. They also ask the contributor to use the same consistent header format that the project uses.

I apologize that we don't have a very clear document about this right now that I can point you to, we're trying to do better in sharing our contribution guidelines and can do better. I'm going to follow up where I can and see what people are working on with our open source office, I bet they have something similar, and we should include it in our repos by default.

If you'd like your name to be in the source code somewhere, you could create a NOTICE file and place it there.

(From a process perspective, we have linter systems in use in some of our teams (a linter is similar to static analysis tools for source files) that look for a consistent header on the source files. This helps us make sure that contributions have either come from direct contributions to the projects (such as yours) or that the inclusion of other software was approved from a team that has had a chance to review which other open source project(s) the contributions came from.)

Does this make sense? If you'd rather contribute this code with no copyright header, that's fine, too, but then we'll have to add that header in to make sure that we don't get any red flags in the future!

**Pull request closed without merging**